lib/puppet/provider/sensuctl_configure/sensuctl.rb

```diff
@@ -66,6 +66,16 @@ def config_namespace=(value)
     @property_flush[:config_namespace] = value
   end
 
+  def backend_init
+    backend = which('sensu-backend')
+    return if backend.nil?
+    return if api.auth_test(resource[:url], resource[:username], resource[:password])
+    return if api.auth_test(resource[:url], resource[:username], resource[:old_password])
+    return if api.auth_test(resource[:url], resource[:username], bootstrap_password)
+    cmd = [backend, 'init', '--cluster-admin-username',resource[:username],'--cluster-admin-password',resource[:password]]
+    execute(cmd, :failonfail => false)
```

Comment:
Need to pass `custom_environment` hash to this execution with `SENSU_BACKEND_CLUSTER_ADMIN_USERNAME ` and `SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD ` and remove the `--cluster-admin-username` and --cluster-admin-password` flags.

Reply:
Sounds like this is still a WIP then.

Reply:
I made this change, this is ready.